WMFComponents/Sources/WMFComponents/Components/Article Tabs/New Tab/WMFNewArticleTabSettingsViewModel.swift

```diff
@@ -0,0 +1,28 @@
+import Foundation
+import Combine
+import WMFData
+
+public final class WMFNewArticleTabSettingsViewModel: ObservableObject {
+    public let title: String
+    public let header: String
+    public let options: [String]
+    public let saveSelection: (Int) -> Void
+
+    @Published public var selectedIndex: Int {
+        didSet {
+            saveSelection(selectedIndex)
+        }
+    }
```

Comment:
(Optional) You do have access to WMFData's userDefaultsStore from here. Instead of passing in this get / set logic in from the outside, You could get and set the user defaults store here directly. Or to match architecture further, call WMFArticleTabsDataController from here which sets the store directly, so that user defaults stores live behind the data controller. It may save you some code at the call sites, but possibly at the cost of making this view model less generic.